packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -159,18 +163,17 @@ class CreateCommand extends CreateBase {
     final bool metadataExists = projectDir.absolute.childFile('.metadata').existsSync();
     if (argResults['template'] != null) {
       template = stringToProjectType(stringArg('template'));
-    } else {
-      // If the project directory exists and isn't empty, then try to determine the template
-      // type from the project directory.
-      if (projectDir.existsSync() && projectDir.listSync().isNotEmpty) {
-        detectedProjectType = determineTemplateType();
-        if (detectedProjectType == null && metadataExists) {
-          // We can only be definitive that this is the wrong type if the .metadata file
-          // exists and contains a type that we don't understand, or doesn't contain a type.
-          throwToolExit('Sorry, unable to detect the type of project to recreate. '
-              'Try creating a fresh project and migrating your existing code to '
-              'the new project manually.');
-        }
+    }
```

Comment:
this could still be within the `else {`, right? since if we've set `template` on line 165 then the value of `detectedProjectType` won't be used anyway, right?

Reply:
If it is in the else, the following code is dead code:

```dart
    if (detectedProjectType != null && template != detectedProjectType && metadataExists) {
      // We can only be definitive that this is the wrong type if the .metadata file
      // exists and contains a type that doesn't match.
      throwToolExit("The requested template type '${flutterProjectTypeToString(template)}' doesn't match the "
          "existing template type of '${flutterProjectTypeToString(detectedProjectType)}'.");
    }
```

`template != detectedProjectType` will never be true.

That means we do not detect inconsistent project types. For example having method channel `plugin` in the .metadata file and `plugin_ffi` as an argument. Not detecting inconsistent project types results in a garbled project state with files from two different templates mixed together.